recipes-app/iot2050-eio-manager/iot2050-eio-manager_0.1.bb

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) Siemens AG, 2023
+#
+# Authors:
+#  Su Bao Cheng <baocheng.su@siemens.com>
+#
+# This file is subject to the terms and conditions of the MIT License.  See
+# COPYING.MIT file in the top-level directory.
+#
+inherit dpkg-raw
+
+DESCRIPTION = "IOT2050 Extended IO Manager"
+MAINTAINER = "baocheng.su@siemens.com"
+
+SRC_URI = " \
+    file://gRPC/EIOManager/iot2050_eio_pb2_grpc.py \
+    file://gRPC/EIOManager/iot2050_eio_pb2.py \
+    file://gRPC/EIOManager/iot2050_eio_pb2.pyi \
+    file://gRPC/EIOManager/iot2050-eio.proto \
+    file://iot2050_eio_global.py \
+    file://iot2050-eio-service.py \
+    file://iot2050-eiod.service \
+    "
+
+DEBIAN_DEPENDS = "python3, python3-grpcio, python3-dotenv"
+
+do_install() {
+    install -v -d ${D}/usr/lib/
+    install -v -d ${D}/usr/lib/iot2050/
+
+    install -v -d ${D}/usr/lib/iot2050/eio
+    install -v -d ${D}/usr/lib/iot2050/eio/gRPC
+    install -v -d ${D}/usr/lib/iot2050/eio/gRPC/EIOManager
+    install -v -m 755 ${WORKDIR}/gRPC/EIOManager/iot2050_eio_pb2_grpc.py ${D}/usr/lib/iot2050/eio/gRPC/EIOManager/
+    install -v -m 755 ${WORKDIR}/gRPC/EIOManager/iot2050_eio_pb2.py ${D}/usr/lib/iot2050/eio/gRPC/EIOManager/
+    install -v -m 755 ${WORKDIR}/gRPC/EIOManager/iot2050_eio_pb2.pyi ${D}/usr/lib/iot2050/eio/gRPC/EIOManager/
+    install -v -m 755 ${WORKDIR}/gRPC/EIOManager/iot2050-eio.proto ${D}/usr/lib/iot2050/eio/gRPC/EIOManager/
+
+    install -v -m 755 ${WORKDIR}/iot2050_eio_global.py ${D}/usr/lib/iot2050/eio/
+    install -v -m 755 ${WORKDIR}/iot2050-eio-service.py ${D}/usr/lib/iot2050/eio/
+
+    install -v -d ${D}/lib/systemd/system/
+    install -v -m 644 ${WORKDIR}/iot2050-eiod.service ${D}/lib/systemd/system/
+
+    install -v -d ${D}/usr/bin/
+    ln -sf ../lib/iot2050/eio/iot2050-eio-service.py ${D}/usr/bin/iot2050-eio-service
+}
```

Comment:
Wouldn't pip packaging be nice for this tool?

Reply:
Any examples?

Reply:
https://packaging.python.org/en/latest/tutorials/packaging-projects/?

Reply:
Just an idea, any given that the debian package later on gains even a binary blob, it may not be the best fit.

Reply:
I plan to make this EIO subsystem a docker image, could be in a separate layer, in the future, so let's keep it as it is now.